great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -532,3 +532,105 @@ def expect_column_values_to_be_unique(self, column, mostly=None,
             having(sa.func.count(sa.column(column)) > 1)
 
         return sa.column(column).notin_(dup_query)
+
+    @MetaSqlAlchemyDataset.column_map_expectation
+    def expect_column_values_to_match_regex(self,
+                                                column,
+                                                regex,
+                                                mostly=None,
+                                                result_format=None, include_config=False, catch_exceptions=None, meta=None
+                                                ):
+        # Postgres-only version
+        if isinstance(self.engine.dialect, sa.dialects.postgresql.dialect):
```

Comment:
I will confess that I don't really know what's happening here. But tests are passing, so I trust it.

Usability question: can users run the *same* expectation_config against pandas and SQLAlchemy with no changes?

Reply:
The issue here is that we're writing sql statements directly (`text`) bypassing the sqlalchemy ORM, but only for dialects that we've specifically added. We can add more dialects as people request them.